### PR TITLE
Added needsPrepare Check and removed error code 99586 check

### PR DESCRIPTION
**Pull request by cheenamalhotra**

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java [hunk 1]
```diff
@@ -69,6 +69,7 @@
     /** True if this execute has been called for this statement at least once */
     private boolean isExecutedAtLeastOnce = false;
 
+    private boolean needsPrepare = false;
```
Comment (TobiasSQL) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: Why do we need `needsPrepare` as a instance field? Should always be local variable.

Comment (cheenamalhotra) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: Fixed.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java [hunk 2]
```diff
@@ -529,6 +528,7 @@ final void doExecutePreparedStatement(PrepStmtExecCmd command) throws SQLServerE
         }
 
         String dbName = connection.getSCatalog();
+        boolean needsPrepare = false;
```
Comment (TobiasSQL) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: Let's make the default `true` to be safe (better prepare than not). Should not matter but just to be a bit defensive...

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java (hunk 1 above)
Comment (TobiasSQL) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: Great!

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java (hunk 2 above)
Comment (TobiasSQL) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: Otherwise things look good :-)

Comment (cheenamalhotra) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: Modified to true.

**Pull request merged**